packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -349,15 +370,48 @@ export class ForkChoice implements IForkChoice {
       this.updateJustified(currentJustifiedCheckpoint, justifiedBalances);
     }
 
-    const targetSlot = computeStartSlotAtEpoch(computeEpochAtSlot(slot));
     const blockRoot = this.config.getForkTypes(slot).BeaconBlock.hashTreeRoot(block);
+    const blockRootHex = toHexString(blockRoot);
+
+    if (this.proposerBoostEnabled && slot === this.fcStore.currentSlot) {
+      let secondsIntoSlot;
+      if (preCachedData?.blockReceptionTime !== undefined) {
+        secondsIntoSlot = (preCachedData.blockReceptionTime - state.genesisTime) % this.config.SECONDS_PER_SLOT;
+        // This will only update fork-choice time into slot
+        this.updateTime(this.fcStore.currentSlot, secondsIntoSlot);
+      } else {
+        secondsIntoSlot = this.fcStore.secondsIntoSlot;
+      }
+
+      // Add proposer score boost if the block is timely
```

Comment:
please consider making a separate function if possible as this `onBlock()` function is already long, for example `computeProposerBoostScore()`

Reply:
Should not be really necessary, the implementation of proposer boost in onBlock is just

```python
    if get_current_slot(store) == block.slot and is_before_attesting_interval:
        store.proposer_boost_root = hash_tree_root(block)
```

https://github.com/ethereum/consensus-specs/pull/2730/files#diff-d4c285a1dd3b0e1fa4b0039ca81142fec96a7e76342b0dc4487883bee593062aR178

Reply:
I have calculated and set score here only, while they calculate the score later: https://github.com/sigp/lighthouse/blob/unstable/consensus/proto_array/src/proto_array.rs#L150 in their `apply_score_changes`.
I can also move score calculation up there.